packages/react/src/components/Dropdown/Dropdown.test.tsx

```diff
@@ -290,262 +243,241 @@ describe('Dropdown', () => {
-      expect(secondItemElement.getAttribute('title')).toEqual('test');
+      const options = getAllByRole('option');
+      expect(options).toHaveLength(6);
 
-      const thirdItemElement = document.querySelector('.ms-Dropdown-item[data-index="3"]') as HTMLElement;
-      expect(thirdItemElement.getAttribute('title')).toBeFalsy();
+      expect(options[0].getAttribute('title')).toBeFalsy();
+      expect(options[1].getAttribute('title')).toEqual('test');
+      expect(options[2].getAttribute('title')).toBeFalsy();
     });
 
     it('opens on focus if openOnKeyboardFocus is true', () => {
-      wrapper = mount(<Dropdown openOnKeyboardFocus label="testgroup" options={DEFAULT_OPTIONS} />);
+      const { getByRole } = render(<Dropdown openOnKeyboardFocus options={DEFAULT_OPTIONS} />);
 
-      wrapper.find('.ms-Dropdown').simulate('focus');
+      userEvent.tab();
 
-      const secondItemElement = document.querySelector('.ms-Dropdown-item[data-index="2"]') as HTMLElement;
-      expect(secondItemElement).toBeTruthy();
+      expect(getByRole('listbox')).toBeTruthy();
     });
 
     it('opens on click if openOnKeyboardFocus is true', () => {
-      wrapper = mount(<Dropdown openOnKeyboardFocus label="testgroup" options={DEFAULT_OPTIONS} />);
+      const { getByRole } = render(<Dropdown openOnKeyboardFocus options={DEFAULT_OPTIONS} />);
 
-      wrapper.find('.ms-Dropdown').simulate('mousedown');
-      wrapper.find('.ms-Dropdown').simulate('click');
+      userEvent.click(getByRole('combobox'));
 
-      const secondItemElement = document.querySelector('.ms-Dropdown-item[data-index="2"]') as HTMLElement;
-      expect(secondItemElement).toBeTruthy();
+      expect(getByRole('listbox')).toBeTruthy();
     });
 
     it('closes on blur when openOnKeyboardFocus is true', () => {
-      wrapper = mount(<Dropdown openOnKeyboardFocus label="testgroup" options={DEFAULT_OPTIONS} />);
+      const { getByRole } = render(<Dropdown openOnKeyboardFocus options={DEFAULT_OPTIONS} />);
 
-      wrapper.find('.ms-Dropdown').simulate('focus');
+      userEvent.tab();
 
-      let dropdownElement = document.querySelector('.ms-Dropdown-items') as HTMLElement;
-      expect(dropdownElement).toBeTruthy();
+      expect(getByRole('listbox')).toBeTruthy();
 
       // blur, force the dropdown to close, then focus again
       // the second focus is simulating the behavior of the Callout when closed
-      wrapper.find('.ms-Dropdown').simulate('blur');
-      wrapper.find('.ms-Dropdown').simulate('keydown', { which: KeyCodes.escape });
-      wrapper.find('.ms-Dropdown').simulate('focus');
+      const dropdownRoot = getByRole('combobox');
+      fireEvent.blur(dropdownRoot);
+      fireEvent.keyDown(dropdownRoot, { which: KeyCodes.escape });
+      fireEvent.focus(dropdownRoot);
 
-      dropdownElement = document.querySelector('.ms-Dropdown-items') as HTMLElement;
-      expect(dropdownElement).toBeFalsy();
+      expect(() => getByRole('listbox')).toThrow();
```

Comment:
Nit: You could use [`queryByRole`](https://testing-library.com/docs/queries/about#types-of-queries), as that will return `null` rather than throw, here to avoid using the arrow function. I find that easier to read but that's 100% subjective :)

Reply:
Oh nice, apparently I didn't read the docs well enough. 😆 I switched to that in a few different places--ones like this where it's expected not to be found, as well as some places where I was checking for the element's existence at the end of the test (and doing getByText/Role with an unnecessary expect.toBeTruthy() afterwards because not expect-ing anything seemed weird).

Before:
```ts
const foo = getByText('whatever');
// silly because getByText would have thrown if the thing was missing
expect(foo).toBeTruthy();
```

After:
```ts
const foo = queryByText('whatever');
expect(foo).toBeTruthy();
```

I still don't love this because the inconsistency about using getBy vs. queryBy bugs me, but at least it gets rid of the redundant assertions. Or I'm open to other suggestions.